thefuck/rules/docker_not_command.py

```diff
@@ -8,24 +8,42 @@
 @sudo_support
 @for_app('docker')
 def match(command):
-    return 'is not a docker command' in command.output
+    return 'is not a docker command' in command.output or 'Usage:	docker' in command.output
 
 
-def get_docker_commands():
-    proc = subprocess.Popen('docker', stdout=subprocess.PIPE)
-    lines = [line.decode('utf-8') for line in proc.stdout.readlines()]
-    lines = dropwhile(lambda line: not line.startswith('Commands:'), lines)
+def _parse_commands(lines, starts_with):
+    lines = dropwhile(lambda line: not line.startswith(starts_with), lines)
     lines = islice(lines, 1, None)
     lines = list(takewhile(lambda line: line != '\n', lines))
     return [line.strip().split(' ')[0] for line in lines]
 
 
+def get_docker_commands():
+    proc = subprocess.Popen('docker', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+    # Old version docker returns its output to stdout, while newer version returns to stderr.
+    lines = proc.stdout.readlines() or proc.stderr.readlines()
+    lines = [line.decode('utf-8') for line in lines]
+
+    # Only newer versions of docker have management commands in the help text.
+    if 'Management Commands:\n' in lines:
+        management_commands = _parse_commands(lines, 'Management Commands:\n')
+    else:
+        management_commands = []
+    regular_commands = _parse_commands(lines, 'Commands:\n')
+    return management_commands + regular_commands
+
+
 if which('docker'):
     get_docker_commands = cache(which('docker'))(get_docker_commands)
 
 
 @sudo_support
 def get_new_command(command):
+    if 'Usage:' in command.output and len(command.script_parts) > 1:
+        management_subcommands = _parse_commands(map(lambda x: x + '\n', command.output.split('\n')), 'Commands:\n')
```

Comment:
I think it might be better to replace `map(lambda x: x + '\n', command.output.split('\n'))` with just `command.output.split('\n')` and replace `lambda line: line != '\n'` in :17 with `lambda line: line.strip()`, so it will handle both `'\n'` and `''`.